annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+    (
+        (
+            MagicMock(
+                user_ids=[2],
+                date_from=datetime(2024, 1, 1),
+                date_to=datetime(2024, 1, 6),
+            ),
+            "annotator_stats_export_20240816.csv",
+        ),
+        (
+            MagicMock(
+                user_ids=[999],
+                date_from=datetime(2024, 1, 1),
+                date_to=datetime(2024, 1, 6),
+            ),
+            "annotator_stats_export_20240816.csv",
+        ),
+    ),
+)
+def test_create_export_csv(
+    schema: Mock,
+    expected_filename: str,
+    mock_db: Mock,
+    mock_user_name: Mock,
+    mock_file_name: Mock,
+):
+    with patch(
+        "annotation.tasks.services.get_user_names_by_user_ids"
+    ) as mock_get_user_names_by_user_ids, patch(
+        "annotation.tasks.services.get_file_names_by_file_ids"
+    ) as mock_get_file_names_by_file_ids, patch(
+        "annotation.tasks.services.io.BytesIO"
+    ) as mock_bytes_io, patch(
+        "annotation.tasks.services.io.TextIOWrapper"
+    ) as mock_text_io_wrapper:
+
+        mock_get_user_names_by_user_ids.return_value = mock_user_name
+        mock_get_file_names_by_file_ids.return_value = mock_file_name
+
+        mock_binary = MagicMock()
+        mock_bytes_io.return_value = mock_binary
+        mock_text_file = MagicMock()
+        mock_text_io_wrapper.return_value = mock_text_file
+
+        filename, _ = create_export_csv(mock_db, schema, "tenant", "token")
+
+        assert filename == expected_filename
+
+
+@pytest.fixture
+def mock_task():
+    task = MagicMock(ManualAnnotationTask)
+    task.id = 1
+    task.user_id = 2
+    task.job_id = 10
+    task.file_id = 3
+    task.pages = {1, 2, 3}
+    task.is_validation = False
+    return task
+
+
+@pytest.fixture
+def mock_tenant_data():
+    mock_data = MagicMock(TenantData)
+    mock_data.token = "mock_token"
+    return mock_data
+
+
+def test_evaluate_agreement_score(
+    mock_session: Mock, mock_task: Mock, mock_tenant_data: Mock
+):
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket"
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_agreement_score"
+    ) as mock_get_agreement_score, patch(
+        "annotation.tasks.services.compare_agreement_scores"
+    ) as mock_compare_agreement_scores:
+
+        mock_get_file_path_and_bucket.return_value = (
+            "mock_s3_file_path",
+            "mock_s3_file_bucket",
+        )
+        mock_agreement_score_response = [
+            MagicMock(AgreementScoreServiceResponse)
+        ]
+        mock_get_agreement_score.return_value = mock_agreement_score_response
+        mock_compare_agreement_scores.return_value = MagicMock(
+            AgreementScoreComparingResult
+        )
+
+        mock_query = MagicMock()
+        mock_query.filter.return_value.filter.return_value.all.return_value = [
+            mock_task
+        ]
+        mock_session.query.return_value = mock_query
```

Comment:
Simplify this chained call. eg:
`mock_session.query().all.return_value = [mock_task]`